### PR TITLE
Added FavoriteActivity, which introduces feature preferences in the app

### PR DESCRIPTION
**Pull request by Marvin99**

### CODE REVIEW

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  setContentView(R.layout.activity_favorite_recipe);
+ }
+
+ @Override
+ public boolean onCreateOptionsMenu(Menu menu) {
+  // Inflate the menu; this adds items to the action bar if it is present.
+  getMenuInflater().inflate(R.menu.favorite_recipe, menu);
+  return true;
+ }
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java:29]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java:30]: this class should  be formatted

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.GridView;
+
+
+public final class FavoriteRecipeFragment extends Fragment {
+    private FavoriteRecipesAdapter adapter;
+    //final ArrayList<FavoriteRecipe> favRecipes = new ArrayList<FavoriteRecipe>();
+
+    //private static Recipe recipe = new Recipe();
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:26]: commented code and empty  line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    final static ArrayList<Recipe> Recipes = new ArrayList<Recipe>();
+    List<Recipe> listFavRecipes = new ArrayList<Recipe>();
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:28]: private

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        return view;
+    }
+
+    @Override
+    public final void onActivityCreated(Bundle savedInstanceState) {
+
+        super.onActivityCreated(savedInstanceState);
+        GridView Recipe = (GridView) getActivity().findViewById(R.id.Recipe);
+        final Recipe recipe = (Recipe) getActivity().getIntent().getSerializableExtra("recipe");
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:45]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        Recipe favRecipe = new Recipe(recipe.getRecipeName(),
+                recipe.getRecipeDetail(), recipe.getCategory() );
+      //  Recipe.add(favRecipe);
+
+        //listFavRecipes = Recipe.getAll();
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:50]: commented code + empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        for(Recipe listFavItem : listFavRecipes){
+            Log.d("DEBUG", "********Fav recipe -> " + listFavItem.getRecipeName());
+            Log.d("DEBUG", "********Fav recipe ING -> " + listFavItem.getRecipeDetail());
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:53]: what does this do ?

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        adapter = new FavoriteRecipesAdapter(getActivity().getBaseContext(), (ArrayList<Recipe>)listFavRecipes);
+        gvFavoriteRecipes.setAdapter(adapter);
+        gvFavoriteRecipes.setOnItemClickListener(new OnItemClickListener() {
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:57]: bad name (gvFavoriteRecipes)

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            @Override
+            public void onItemClick(AdapterView<?> adapter, View parent,
+                                    int position, long rowId) {
+                Recipe imageResult = listFavRecipes.get(position);
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:62]: empty line and final everywhere

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                Recipe r = new Recipe(imageResult.getRecipeName(), imageResult.getRecipeHowToCook(), imageResult.getCategory());
+                Intent i = new Intent(getActivity(), RecipeDetailsActivity.class);
+               // i.putExtra("result", r);
+                startActivity(i);
+            }
+        });
+
+    }
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:71]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public final View getView(int position, View convertView, ViewGroup parent) {
+        Recipe favRecipe = this.getItem(position);
+        RelativeLayout rlRecipe;
+        ImageView svImageRecipe;
+        TextView tvRecipeName;
+
+        if(convertView == null){
+            //LayoutInflater inflater = LayoutInflater.from(getContext());
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:31]: commented code

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+            rlRecipe = (RelativeLayout) inflater.inflate(R.layout.item_recipe, parent, false);
+
+        } else {
+            rlRecipe = (RelativeLayout) convertView;
+        }
+
+        svImageRecipe = (ImageView) rlRecipe.findViewById(R.id.ivRecipe);
+        tvRecipeName = (TextView) rlRecipe.findViewById(R.id.tvName);
+        //svImageRecipe.setImageUrl(favRecipe.getRecipeUrl());
+        //Log.d("debug", "URL  : " + favRecipe.getRecipeUrl());
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:42]: commented code

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        tvRecipeName.setText(favRecipe.getRecipeName());
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:44]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        return rlRecipe;
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:46]: empty line

File: app/src/main/java/bubbleheads/buy_cook/RecipeDetailsActivity.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+        FragmentTransaction transaction=manager.beginTransaction();//create an instance of Fragment-transaction
+
+        transaction.add(R.id.flTopFragment, topFragment, "Frag_Top_tag");
+        transaction.add(R.id.flMiddleFragment, middleFragment, "Frag_Bottom_tag");
+        //transaction.add(R.id.flBottomFragment, bottomFragment, "Frag_Middle_tag");
+
+        transaction.commit();*/
+    }
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/RecipeDetailsActivity.java:48]: commented code

File: app/src/main/java/bubbleheads/buy_cook/RecipeDetailsActivity.java [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                // app icon in action bar clicked; go home
+                //Intent intent = new Intent(this, HomeActivity.class);
+                //intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                //startActivity(intent);
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/RecipeDetailsActivity.java:56]: commented

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 17]
```diff
@@ -0,0 +1,72 @@
+package bubbleheads.buy_cook;
+
+/**
+ * Created by MARVIN on 07.04.2017.
+ */
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:6]: it should be deleted

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+    public final View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // TODO Auto-generated method stub
+        View view = inflater.inflate(R.layout.fragment_favorite_recipe_activity, container, false);
+        return view;
+    }
+
+    @Override
+    public final void onActivityCreated(Bundle savedInstanceState) {
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:41]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+    //final ArrayList<FavoriteRecipe> favRecipes = new ArrayList<FavoriteRecipe>();
+
+    //private static Recipe recipe = new Recipe();
+    final static ArrayList<Recipe> Recipes = new ArrayList<Recipe>();
+    List<Recipe> listFavRecipes = new ArrayList<Recipe>();
+    private GridView gvFavoriteRecipes;
+
+    @Override
+    public final View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:33]: inflater,container and savedInstanceState should be final

File: app/src/main/java/bubbleheads/buy_cook/RecipeDetailedFragment.java [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+
+public final class RecipeDetailedFragment extends Fragment {
+    private Recipe detailedRecipe;
```
Comment (nastusya) [inline app/src/main/java/bubbleheads/buy_cook/RecipeDetailedFragment.java:13]: you don`t need this field. You can do this like a local variable

File: app/src/main/java/bubbleheads/buy_cook/RecipeDetailsActivity.java [hunk 21]
```diff
@@ -0,0 +1,62 @@
+package bubbleheads.buy_cook;
+
+/**
+ * Created by MARVIN on 07.04.2017.
+ */
```
Comment (nastusya) [inline app/src/main/java/bubbleheads/buy_cook/RecipeDetailsActivity.java:5]: Delete this commented code

File: app/src/main/res/layout/detailed_recipe_fragment.xml [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+            android:id="@+id/floatingActionButton2"
+            android:layout_alignParentBottom="true"
+            android:layout_toStartOf="@+id/floatingActionButton3"
+            android:layout_marginEnd="24dp"
+            android:layout_margin="13dp"
+            app:elevation="0dp"
+            android:paddingStart="2dp" />
+
+    </RelativeLayout>
+</ScrollView>
```
Comment (nastusya) [inline app/src/main/res/layout/detailed_recipe_fragment.xml:96]: please, reformat your code and all values should be divided by 8

File: app/src/main/java/bubbleheads/buy_cook/RecipeDetailedFragment.java [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+
+public final class RecipeDetailedFragment extends Fragment {
```
Comment (nastusya) [inline app/src/main/java/bubbleheads/buy_cook/RecipeDetailedFragment.java:12]: this class shouldn't be final

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+        adapter = new FavoriteRecipesAdapter(getActivity().getBaseContext(), (ArrayList<Recipe>)listFavRecipes);
+        gvFavoriteRecipes.setAdapter(adapter);
+        gvFavoriteRecipes.setOnItemClickListener(new OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapter, View parent,
+                                    int position, long rowId) {
+                Recipe imageResult = listFavRecipes.get(position);
+
+                Recipe r = new Recipe(imageResult.getRecipeName(), imageResult.getRecipeHowToCook(), imageResult.getCategory());
+                Intent i = new Intent(getActivity(), RecipeDetailsActivity.class);
```
Comment (nastusya) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:64]: What means r and i? it`s bad name for variables

File: app/src/main/res/layout/detailed_recipe_fragment.xml [hunk 25]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <RelativeLayout
```
Comment (MartaLiutyk) [inline app/src/main/res/layout/detailed_recipe_fragment.xml:7]: do you need RelativeLayout here ?

File: app/src/main/res/layout/detailed_recipe_fragment.xml [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+        android:orientation="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
```
Comment (MartaLiutyk) [inline app/src/main/res/layout/detailed_recipe_fragment.xml:11]: empty line

File: app/src/main/res/layout/fragment_favorite_recipe_activity.xml [hunk 27]
```diff
@@ -0,0 +1,17 @@
...
+        android:id="@+id/gvFavoriteRecipes"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentTop="true"
+        android:numColumns="2"
+        android:columnWidth="400dp" >
+    </GridView>
+
+
```
Comment (MartaLiutyk) [inline app/src/main/res/layout/fragment_favorite_recipe_activity.xml:16]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java [hunk 28]
```diff
@@ -0,0 +1,30 @@
+package bubbleheads.buy_cook;
+
+/**
+ * Created by MARVIN on 07.04.2017.
+ */
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java:6]: this should be deleted

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.ArrayList;
+
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+import android.view.Menu;
+import android.widget.GridView;
+import android.widget.Toast;
+
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java:15]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.GridView;
+
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:21]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+                                    int position, long rowId) {
+                Recipe imageResult = listFavRecipes.get(position);
+
+                Recipe r = new Recipe(imageResult.getRecipeName(), imageResult.getRecipeHowToCook(), imageResult.getCategory());
+                Intent i = new Intent(getActivity(), RecipeDetailsActivity.class);
+               // i.putExtra("result", r);
+                startActivity(i);
+            }
+        });
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeFragment.java:69]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 32]
```diff
@@ -0,0 +1,48 @@
+package bubbleheads.buy_cook;
+
+/**
+ * Created by MARVIN on 07.04.2017.
+ */
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:6]: this should be deleted

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 33]
```diff
@@ -0,0 +1,48 @@
...
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+public final class FavoriteRecipesAdapter extends ArrayAdapter<Recipe> {
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:18]: this class shouldn`t be final

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 34]
```diff
@@ -0,0 +1,48 @@
...
+    public FavoriteRecipesAdapter(Context context, ArrayList<Recipe> favRecipes) {
+        super(context, 0, favRecipes);
+    }
+
+    @Override
+    public final View getView(int position, View convertView, ViewGroup parent) {
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:24]: method shouldn`t  be final, parameters should be  final

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java [hunk 35]
```diff
@@ -0,0 +1,48 @@
...
+        Recipe favRecipe = this.getItem(position);
+        RelativeLayout rlRecipe;
+        ImageView svImageRecipe;
+        TextView tvRecipeName;
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipesAdapter.java:29]: bad names for variables, in method some of them should be final

File: app/src/main/java/bubbleheads/buy_cook/RecipeDetailedFragment.java [hunk 36]
```diff
@@ -0,0 +1,30 @@
...
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+
```
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/RecipeDetailedFragment.java:11]: empty line

File: app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java [hunk 37]
```diff
@@ -0,0 +1,30 @@
...
+public class FavoriteRecipeActivity extends FragmentActivity {
```
Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/FavoriteRecipeActivity.java:16]: there is no need in another activity

**Review by ovovchak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**